socketFunctions.js

```diff
@@ -0,0 +1,52 @@
+const axios = require('axios');
+const crypto = require("crypto");
+
+const { addUser, removeUser, getUser, getUsersInRoom } = require('./users');
+
+const onJoin = (socket, io) => {
+    socket.on('join', async ({ name, room }, callback) => {
+        const { error, user } = addUser({ id: socket.id, name, room });
+        socket.join(user.room);
+        if (error) return callback(error);
+
+        const id = crypto.randomBytes(16).toString("hex");
+        socket.emit('message', { id: id, sender: 'admin', text: `${user.name}, welcome to room ${user.room}` });
+        socket.broadcast.to(user.room).emit('message', { id: id, sender: 'admin', text: `${user.name} has joined` });
+        io.to(user.room).emit('roomData', { room: user.room, users: getUsersInRoom(user.room) });
+        
+        if (getUsersInRoom(user.room).length === 2) {
+            try {
+                const response = await axios.get('https://uselessfacts.jsph.pl/random.json?language=en');
+                socket.to(user.room).emit('message', { id: response.data.id, sender: 'Random Fact Teller', text: response.data.text });
+                socket.emit('message', { id: response.data.id, sender: 'Random Fact Teller', text: response.data.text });
```

Comment:
Because the data is the same for this `to` and `emit`, what is the difference here?